pkg/forwarder/failed_transaction_removal_policy.go

```diff
@@ -18,21 +18,34 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util"
 )
 
+type failedTransactionRemovalPolicyTelemetry interface {
```

Comment:
What's the need to go through an interface here?

Directly using `retryQueueTelemetry` in `type failedTransactionRemovalPolicy struct` was an issue?

Reply:
Using an interface allows the unit tests to not have a side effect (The implementation of this interface uses globals behind the scene). I had bad story of debugging unit tests because of a side effect of the previous unit tests side effect.

Note: I started to check the telemetry in the unit tests but decided to not do it at the end.